numba/tests/test_errorhandling.py

```diff
@@ -453,14 +453,14 @@ class TestDeveloperSpecificErrorMessages(SerialMixin, unittest.TestCase):
     def test_bound_function_error_string(self):
         # See PR #5952
         def foo(x):
-            x.max(-1) # axis not supported
+            x.max(-1)
 
         with override_config('DEVELOPER_MODE', 1):
             with self.assertRaises(errors.TypingError) as raises:
                 njit("void(int64[:,:])")(foo)
 
         excstr = str(raises.exception)
-        self.assertIn("args not supported", excstr)
+        self.assertIn("too many positional arguments", excstr)
```

Comment:
Originally introduced in https://github.com/numba/numba/pull/5952.

This error message was later changed but it deviated from it's intended purpose. i.e. As of now it was verifying the comment made after the function rather than the function itself.

Reply:
Not to resolve in this PR... but this error message seems less specific/clear about what the problem is. "args not support" is relatively clear, "too many positional arguments" requires some inference about how many positional args there are in the user code and how that might match what is supported (which also isn't clear). It also seems like the exception handling on typing failure for these bound functions might need some customisation as `array.max()` is being wired through to `np.max()`.